### PR TITLE
[clusteragent/autoscaling] Use `PodWatcher` to update current replicas in status

### PR DESCRIPTION
**Pull request by jennchenn**
### What does this PR do?

Use `PodWatcher` to update current replica count in `DatadogPodAutoscaler` status  instead of Horizontal `/scale` sub-resource.

### Motivation

If horizontal scaling has no changes/is not activated, the current replica count may be inaccurate. This change moves the logic to update the number of current replicas to the controller loop to ensure it is as up-to-date as possible.

### Possible Drawbacks / Trade-offs

During testing, sometimes some lag between when `PodWatcher` information is updated and when a scaling action happens exists. This can lead to a maximum of 5 min of inconsistent number of replicas being reported (info is fixed on next run of the controller loop). 

~~There is currently duplicated logic between the controller and vertical controller to get the number of pods - would it make sense to combine these two to avoid calling `podWatcher.GetPodsForOwner` twice? The now duplicated logic:
https://github.com/DataDog/datadog-agent/blob/2f3053a37cbe4c1fc2327763f39e9bef1cac3ab1/pkg/clusteragent/autoscaling/workload/controller_vertical.go#L70-L84~~

### Describe how to test/QA your changes

1. Run an autoscaling workload
2. Verify that the `Current Replicas` count in the status is updated after scaling actions (even if horizontal scaling is disabled)

### CODE REVIEW

File: pkg/clusteragent/autoscaling/workload/controller.go [hunk 1]
```diff
@@ -275,6 +275,20 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 	// Now that everything is synced, we can perform the actual processing
 	result, err := c.handleScaling(ctx, podAutoscaler, &podAutoscalerInternal)
 
+	// Update current replicas
+	targetGVK, targetErr := podAutoscalerInternal.TargetGVK()
+	if targetErr != nil {
+		podAutoscalerInternal.SetError(targetErr)
+		return autoscaling.NoRequeue, targetErr
```
Comment (vboulineau) [inline pkg/clusteragent/autoscaling/workload/controller.go:282]: In that case you'd miss status update

**Review by vboulineau — COMMENTED** [review]
Yes we should dedupe the code and give the necessary info to vertical controller.

File: pkg/clusteragent/autoscaling/workload/controller.go (hunk 1 above)
Comment (jennchenn) [reply pkg/clusteragent/autoscaling/workload/controller.go]: Added a check to update status and return the error at the end; priority for error during scaling > error to get target > error when updating status

Comment (jennchenn) [reply pkg/clusteragent/autoscaling/workload/controller.go]: Makes sense, I can wait for the changes from https://github.com/DataDog/datadog-agent/blob/f35a065b1e4c9f735ea00edceba0718f249bbe5a/pkg/clusteragent/autoscaling/workload/controller.go#L413-L427
to be merged so I can make a status update and return early once `targetErr` is encountered

**Review by sblumenthal — APPROVED** [review]

**Pull request merged**